nautobot/docs/rest-api/overview.md

```diff
@@ -785,3 +785,478 @@ http://nautobot/api/dcim/sites/ \
+
+### Depth Query Parameter
+
++++ 2.0.0
+
+The the built-in DRF `?depth` query parameter is introduced in Nautobot 2.0 to replace the `?brief` parameter. It enables [nested serialization](https://www.django-rest-framework.org/api-guide/serializers/#specifying-nested-serialization) functionality and offers a more dynamic and comprehensive browsable API.
+This parameter is an positive integer value that can range from 0 to 10. In most use cases, you will only need a maximum `depth` of 2 to get all the information you need.
+
+!!! important
+    The `?depth` query parameter should only be used for `GET` operations in the API. It should not be used in `POST`, `PATCH` and `DELETE` requests. For these requests, only `?depth=0` should be used.
+
+#### Default/?depth=0
+
+`?depth` parameter is default to 0 and offers a very simplistic view of the API where all object-related fields are represented by only their primary keys:
+
+```no-highlight
+curl -s -X GET \
+-H "Accept: application/json; version=2.0" \
+http://nautobot/api/dcim/locations/0e19e475-89c9-4cf4-8b5f-a0589f0950cd/ | jq '.'
+```
+
+```json
+{
+    "id": "0e19e475-89c9-4cf4-8b5f-a0589f0950cd",
+    "display": "Campus-01",
+    "url": "http://nautobot/api/dcim/locations/0e19e475-89c9-4cf4-8b5f-a0589f0950cd/",
+    "custom_fields": {
+        "example_plugin_auto_custom_field": null
+    },
+    "notes_url": "http://nautobot/api/dcim/locations/0e19e475-89c9-4cf4-8b5f-a0589f0950cd/notes/",
+    "tree_depth": 0,
+    "time_zone": "Asia/Baghdad",
+    "circuit_count": 7,
+    "device_count": 0,
+    "prefix_count": 0,
+    "rack_count": 0,
+    "virtual_machine_count": 0,
+    "vlan_count": 0,
+    "created": "2023-04-12T19:29:06.884754Z",
+    "last_updated": "2023-04-12T19:29:06.906503Z",
+    "name": "Campus-01",
+    "slug": "campus-01",
+    "description": "Local take each compare court exactly.",
+    "facility": "328",
+    "asn": null,
+    "physical_address": "",
+    "shipping_address": "",
+    "latitude": null,
+    "longitude": "104.200000",
+    "contact_name": "Frances Hernandez",
+    "contact_phone": "",
+    "contact_email": "",
+    "comments": "Sort share road candidate.",
+    "status": "28eb334b-4171-4da4-a03a-fa6d0c6a9442",
+    "parent": null,
+    "location_type": "e3d4a9af-c6c1-4582-b483-a13301eb6e28",
+    "tenant": "5b1feadb-fab0-4f81-a53f-5192d83b0216",
+    "tags": [
+        "6e8ce6c9-0a8c-4731-b02a-b2ec19db5c52",
+        "1e986e5d-099d-4742-bfa5-16363abea5fd",
+        "21afbdad-c782-4fc5-9b3b-7b59273bb08c",
+        "0118b994-9943-4295-946b-92b27efe15db"
+    ]
+}
+```
+
+#### ?depth=1
+
+When `?depth=1` is added to the query parameter, all object-related fields, instead of being represented as UUIDs as they are when `?depth=0`, will be replaced by the related models' serializers.
```

Comment:
```suggestion
When `?depth=1` is added to the query parameters, all object-related fields, instead of being represented as UUIDs as they are when `?depth=0`, will be replaced by the related models' serializers.
```